tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Kinesis/GetRecordsIntegration.cs

```diff
@@ -0,0 +1,99 @@
+// <copyright file="GetRecordsIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DataStreamsMonitoring;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Propagators;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.Kinesis
+{
+    /// <summary>
+    /// AWSSDK.Kinesis GetRecords CallTarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "AWSSDK.Kinesis",
+        TypeName = "Amazon.Kinesis.AmazonKinesisClient",
+        MethodName = "GetRecords",
+        ReturnTypeName = "Amazon.Kinesis.Model.GetRecordsResponse",
+        ParameterTypeNames = new[] { "Amazon.Kinesis.Model.GetRecordsRequest" },
+        MinimumVersion = "3.0.0",
+        MaximumVersion = "3.*.*",
+        IntegrationName = AwsKinesisCommon.IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class GetRecordsIntegration
+    {
+        private const string Operation = "GetRecords";
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TGetRecordsRequest">Type of the request object</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method</param>
+        /// <param name="request">The request for the Kinesis operation</param>
+        /// <returns>CallTarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TGetRecordsRequest>(TTarget instance, TGetRecordsRequest request)
+            where TGetRecordsRequest : IGetRecordsRequest, IDuckType
+        {
+            if (request.Instance is null)
+            {
+                return CallTargetState.GetDefault();
+            }
+
+            var scope = AwsKinesisCommon.CreateScope(Tracer.Instance, Operation, SpanKinds.Producer, null, out var tags);
+
+            string? streamName = AwsKinesisCommon.StreamNameFromARN(request.StreamARN);
```

Comment:
Same comment as above about skipping `AwsKinesisCommon.StreamNameFromARN()` is not needed.